### PR TITLE
Armory Release Workflow 

### PR DESCRIPTION
**Pull request by christopherwoodall**
Updates Armory's release workflow.

### CODE REVIEW

File: .github/actions/environment_setup/action.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    required: true
+    default: "3.9"
+
+runs:
+  using: "composite"
+  steps:
+    - name: 🐍 Switch Python ${{ inputs.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ inputs.python-version }}
```
Comment (mwartell) [inline .github/actions/environment_setup/action.yml:17]: This is one case where parameterizing "3.9" creates a ton of syntactic junk. I'm not sure this is a win.

Better might be an encapsulation (refactor) of this whole workflow block
```
    name: ☁️ Pytorch Image Tests [Container]
    runs-on: ubuntu-latest
    steps:
      - name: 🐄 checkout armory full depth with tags for scm
        uses: actions/checkout@v3
        with:
          fetch-depth: 0

      # Get history and tags for SCM versioning to work
      # yamllint disable rule:indentation
      - name: 📩 Checkout the Source
        run: |
          git fetch --prune --unshallow || true
          git fetch --depth=1 origin +refs/tags/*:refs/tags/*
          # Verify .git folder
          if [ ! -d "./.git" ]; then
            echo "Expected ./.git folder to exist"
            exit 1
          fi
          python -m setuptools_scm
          pwd
          git log --oneline --decorate -n 1
      # yamllint enable rule:indentation

      - name: 🐍 Use Python 3.9
        uses: actions/setup-python@v4
        with:
          python-version: 3.9
```

Comment (christopherwoodall) [reply .github/actions/environment_setup/action.yml]: That action is not being used as of yet so I just removed the entire directory.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,96 +1,141 @@
...
+            echo "Expected ./.git folder to exist"
+            exit 1
+          fi
+
+      - name: 🔨 Build wheel
         run: |
-          pip install wheel
-          python setup.py bdist_wheel
-      - uses: pypa/gh-action-pypi-publish@master
+          pip install pip==22.2.2
```
Comment (mwartell) [inline .github/workflows/release.yml:54]: pip is moving way to fast this year to pin to a dot release, better would be `>=`

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -1,96 +1,142 @@
...
-          pip install -r requirements.txt
-          version=$(python -m armory --version)
-          docker login --username ${{ secrets.DOCKER_USERNAME }} --password ${{ secrets.DOCKER_PASSWORD }}
-          docker pull twosixarmory/pytorch-deepspeech:latest
-          python docker/build.py pytorch-deepspeech
-          docker push twosixarmory/pytorch-deepspeech:${version}
-          docker tag twosixarmory/pytorch-deepspeech:${version} twosixarmory/pytorch-deepspeech:latest
-          docker push twosixarmory/pytorch-deepspeech:latest
+          echo "Building Armory ${{ env.ARMORY_VERSION }} Images"
+          python docker/build.py --framework all
```
Comment (davidslater) [inline .github/workflows/release.yml:118]: Previously, this caused OOM errors on the runner.

Comment (christopherwoodall) [reply .github/workflows/release.yml]: Updated to use matrix.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -1,96 +1,172 @@
...
-          pip install wheel
-          python setup.py bdist_wheel
-      - uses: pypa/gh-action-pypi-publish@master
+          git fetch --prune --unshallow || true
+          git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+          # Verify .git folder
+          if [ ! -d "./.git" ]; then
+            echo "Expected ./.git folder to exist"
+            exit 1
+          fi
```
Comment (mwartell) [inline .github/workflows/release.yml:54]: This is likely unneeded per https://github.com/twosixlabs/armory/pull/1646

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -1,96 +1,172 @@
...
-    runs-on: ubuntu-18.04
-    steps:
-      - uses: actions/checkout@v2
+          git fetch --prune --unshallow || true
+          git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+          # Verify .git folder
+          if [ ! -d "./.git" ]; then
+            echo "Expected ./.git folder to exist"
+            exit 1
+          fi
```
Comment (mwartell) [inline .github/workflows/release.yml:114]: This is likely unneeded per https://github.com/twosixlabs/armory/pull/1646

**Review by mwartell — APPROVED** [review]
I walked through this with you. LGTM modulo my aesthetics comments.

**Comment by mwartell** [comment]
Closed by #1653

**Pull request closed without merging**